### PR TITLE
[N/A] First Pass Review Updates

### PR DESCRIPTION
**Pull request by rypete**
Opening a PR for this to highlight changes made.  

Brief: 
* Update flag help docs to provide possible options.
* Fixed missing acorn dependency
* Split CLI out from the index to fix importing issues.  This is now under cli.ts and is used by .bin
* Moved middleware to live under the server directory as it was mostly express related
* Rename the webpack export to be `webpackTools`
* Renamed webpack middleware to `executeWebpack` - this still acts as both middleware and standard webpack executor

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * Start command
+ */
+program.command('start')
+    .option(
+        '-v, --providerVersion <version>',
+        'Sets the runtime version for the provider.  Defaults to "local". Options: local | staging | stable | w.x.y.z',
+        'local')
+    .option('-m, --mode <mode>', 'Sets the webpack mode.  Defaults to "development".  Options: development | production | none', 'development')
+    .option('-l, --noDemo', 'Runs the server but will not launch the demo application.', true)
```
Comment (pjbroadbent) [inline src/cli.ts]: Can we rename the short flag?

`-n`? `-d`?..

File: src/webpack/webpack.ts [hunk 2]
```diff
@@ -1,104 +1,48 @@
...
-    return new CopyWebpackPlugin([{
-        from: './res/provider/app.json',
-        to: '.',
-        transform: (content) => {
-            const config = JSON.parse(content);
-
-            if (typeof process.env.SERVICE_VERSION !== 'undefined' && process.env.SERVICE_VERSION !== '') {
-                config.startup_app.url = `https://cdn.openfin.co/services/openfin/${SERVICE_NAME}/` + process.env.SERVICE_VERSION + '/provider.html';
-                config.startup_app.autoShow = false;
+export async function executeWebpack(mode: WebpackMode, writeToDisk: boolean): Promise<RequestHandler> {
```
Comment (pjbroadbent) [inline src/webpack/webpack.ts]: This is now the only thing in here, rename file to `executeWebpack.ts`?

File: package.json [hunk 3]
```diff
@@ -1,37 +1,38 @@
 {
   "name": "openfin-service-tooling",
-  "version": "1.0.0",
+  "version": "1.0.2",
```
Comment (pjbroadbent) [inline package.json]: Are you incrementing this on each commit? It should only be incremented on each release, and the module hasn't been published yet.

File: src/cli.ts (hunk 1 above)
Comment (rypete) [reply src/cli.ts]: Going with `-n`

File: src/webpack/webpack.ts (hunk 2 above)
Comment (rypete) [reply src/webpack/webpack.ts]: Good catch, fixed

File: package.json (hunk 3 above)
Comment (rypete) [reply package.json]: I was supposed to set this back to 0 before push, whoops.  Fixed now

**Review by pjbroadbent — APPROVED** [review]

**Pull request merged**